superbench/benchmarks/micro_benchmarks/directx_utils/Options.h

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+class Options {
+  protected:
+    char **begin;
+    char **end;
+
+    /**
+     * @brief Get the char* value of the cmd line argument.
+     * @param option the argument in cmd.
+     * @return char*
+     */
+    char *get_cmd_option(const std::string &option) {
+        char **itr = std::find(begin, end, option);
+        if (itr != end && ++itr != end) {
+            return *itr;
+        }
+        return 0;
+    }
+
+    /**
+     * @brief Get the int type value of cmd line argument.
+     * @param option the cmd line argument.
+     * @param defaults the default value.
+     * @return int the int type value of cmd line argument 'option'.
+     */
+    int get_cmd_line_argument_int(const std::string &option, int defaults) {
+        if (char *value = get_cmd_option(option)) {
+            try {
+                return std::stoi(value);
+            } catch (const std::exception &e) {
```

Comment:
well, if there is exception, you are going to return the default value, right? If it is so, I guess you can return defaults directly in the catch clause instead of using exit(1), but if you need to quit the whole process caused by that, you can keep it 😊